### PR TITLE
support mysql create and drop view refresh meta data

### PR DESCRIPTION
**Pull request by strongduanmu**
Ref #7138.

Changes proposed in this pull request:
- support mysql create and drop view refresh meta data
- modify routing way of mysql function and procedure statements to all databases route

### CODE REVIEW

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-route/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/impl/ShardingCreateProcedureStatementValidator.java [hunk 1]
```diff
@@ -42,7 +44,9 @@ public void preValidate(final ShardingRule shardingRule, final SQLStatementConte
         routineBodySegment.ifPresent(routineBody -> {
             TableExtractor extractor = new TableExtractor();
             validateTableNotExist(metaData, extractor.extractNotExistTableFromRoutineBody(routineBody));
-            validateShardingTable(metaData, extractor.extractExistTableFromRoutineBody(routineBody));
+            Collection<SimpleTableSegment> existTables = extractor.extractExistTableFromRoutineBody(routineBody);
+            validateShardingTable(metaData, existTables);
```
Comment (tristaZero) [inline shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-route/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/impl/ShardingCreateProcedureStatementValidator.java:48]: Hi @strongduanmu 
I know this PR is a time-consuming work, Very appreciated your initiative!
`Create procedure, function, view` is a

**Review by tristaZero — CHANGES_REQUESTED** [review]
😃

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-route/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/impl/ShardingCreateProcedureStatementValidator.java (hunk 1 above)
Comment (strongduanmu) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-route/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/impl/ShardingCreateProcedureStatementValidator.java:48]: > Hi @strongduanmu
> I know this PR is a time-consuming job, Very appreciated your initiative!
> `Create procedure, function, view` is a troublesome but user-liked issue that sometimes frustrates me to think about it. You are the pioneer for this challenging one, so I wish you could improve it step by step for a satisfying solution.
> At present, I can see the `create view` is narrowed to those single tables (no sharding tables, broadcast tables) , isn't it? It seems a critical first step for me, and I will give my +1 vote.
> However, here It looks like you enable `create function, procedure` on the sharding table as well, right? Do you think these SQL could work well on all the sharding tables and single tables? Or are there any restrictions?
> 
> Look forward to hearing from you.
> 
> Best,
> Trista

@tristaZero The `create function` and `create procedure` statements have previously added a validator to limit the use of the sharding table. This pr is only to modify the routing way of the `create function` and `create procedure` statements and change it to all databases routing. 😂

**Review by tristaZero — APPROVED** [review]

**Pull request merged**